diagrams/ccms-ebs-containers.yaml

```
@@ -0,0 +1,88 @@
+links:
+  The FC4 Framework: https://fundingcircle.github.io/fc4-framework/
+  Structurizr Express: https://structurizr.com/express
+---
+type: Container
+scope: Client Cost Management System (CCMS)
+description: The Provider User Interface within the Client Cost Management System. Omits details of the Provider UI components.
+
+elements:
+- type: Person
+  name: Caseworker
+  description: Civil Legal Aid Caseworker
+  position: '1425,50'
+- type: Person
+  name: Provider
+  description: Civil Legal Aid Provider
+  position: '825,50'
+- type: Software System
+  name: Client Cost Management System (CCMS)
+  containers:
+  - type: Container
+    name: (Logical) Provider User Interface
+    description: Collects information for a legal aid application
+    position: '600,1100'
+  - type: Container
+    name: Oracle E-Business Suite
+    position: '2000,1100'
+    description: Stores completed applications and provides a workflow for accepting or rejecting them.
+  - type: Container
+    name: Oracle Service Oriented Architecture
+    description: Connects EBS to other systems.
+    position: '1300,1100'
+- type: Software System
+  name: Oracle Access Manager
+  position: '100,600'
+- type: Software System
+  name: Oracle HTTP Server
+  position: '800,600'
+- type: Software System
+  name: Portal
+  position: '1900,600'
+
+relationships:
+- source: (Logical) Provider User Interface
+  description: Submit application
+  destination: Oracle Service Oriented Architecture
+- source: Caseworker
+  description: Amend applications
+  destination: Oracle HTTP Server
+- source: Caseworker
+  description: Assess applications
+  destination: Portal
+- source: Oracle HTTP Server
+  description: Proxies requests to
+  destination: (Logical) Provider User Interface
+- source: Oracle HTTP Server
+  description: Authenticates users using
+  destination: Oracle Access Manager
```

In container diagrams, I think it's useful to focus on the system; is it useful to focus on Portal's internals? I.e. could we write

```suggestion
  destination: Portal
```

?

---

Or to spin the question around, I'm confused where does this access manager belong. If it's CCMS I think it should be inside the system boundary, if it's part of another system I feel it should have that system's name.

---

That's a good question - I'm not sure how to draw that boundary because I think we might configure OAM for Provider UI (part of CCMS) but I also recall hearing that other systems use OAM or share data with it. @jabley are you able to shed any light on this? I agree it makes sense to remove these bits if they are part of portal.

---

After talking to @davidkelliott I understand the authentication flow a bit better but I need to think about what this diagram should show.

OAM is definitely part of Portal. So for the purposes of this diagram we can just call it portal. A slight complication is that EBS shares user data with OAM - both are populated by a hub job. 

My understanding is that both caseworkers and providers log in via portal. When a request comes into the app it hits either AccessGate (for EBS) or WebGate (for PUI). WebGate is part of OHS and AccessGate is a standalone component. These check an authentication cookie that is set when you log in via portal.

I think this container diagram should just show Portal outside of the system boundary, and Logical PUI, EBS/SOA, and OHS inside of the system boundary.

---

👍 I like the idea of showing "Portal" as a single node.

💭 In #15, I've used `via-hub` tags on relationships to show "hub" as a protocol, rather than a system; I think it works well for narrative purposes.